src/cascadia/TerminalApp/AppCommandlineArgs.cpp

```diff
@@ -0,0 +1,506 @@
+            {
+                return _handleExit(*_newTabCommand, e);
+            }
+        }
+        else
+        {
+            return _handleExit(_app, e);
+        }
+    }
+    return 0;
+}
+
+// Method Description:
+// - Calls App::exit() for the provided command, and collects it's output into
+//   our _exitMessage buffer.
+// Arguments:
+// - command: Either the root App object, or a subcommand for which to call exit() on.
+// - e: the CLI::Error to process as the exit reason for parsing.
+// Return Value:
+// - 0 if the command exited successfully
+int AppCommandlineArgs::_handleExit(const CLI::App& command, const CLI::Error& e)
+{
+    // Create some streams to collect the output that would otherwise go to stdout.
+    std::ostringstream out;
+    std::ostringstream err;
+    const auto result = command.exit(e, out, err);
+    // I believe only CallForHelp will return 0
+    if (result == 0)
+    {
+        _exitMessage = out.str();
+    }
+    else
+    {
+        _exitMessage = err.str();
+    }
+    return result;
+}
+
+// Method Description:
+// - Add each subcommand and options to the commandline parser.
+// Arguments:
+// - <none>
+// Return Value:
+// - <none>
+void AppCommandlineArgs::_buildParser()
+{
+    _buildNewTabParser();
+    _buildSplitPaneParser();
+    _buildFocusTabParser();
+}
+
+// Method Description:
+// - Adds the `new-tab` subcommand and related options to the commandline parser.
+// Arguments:
+// - <none>
+// Return Value:
+// - <none>
+void AppCommandlineArgs::_buildNewTabParser()
+{
+    _newTabCommand = _app.add_subcommand("new-tab", "Create a new tab");
+    _addNewTerminalArgs(_newTabCommand);
+    _newTabCommand->callback([&, this]() {
+        // Buld the NewTab action from the values we've parsed on the commandline.
+        auto newTabAction = winrt::make_self<implementation::ActionAndArgs>();
+        newTabAction->Action(ShortcutAction::NewTab);
+        auto args = winrt::make_self<implementation::NewTabArgs>();
+        args->TerminalArgs(_getNewTerminalArgs());
+        newTabAction->Args(*args);
+        _startupActions.push_back(*newTabAction);
+    });
+}
+
+// Method Description:
+// - Adds the `split-pane` subcommand and related options to the commandline parser.
+// Arguments:
+// - <none>
+// Return Value:
+// - <none>
+void AppCommandlineArgs::_buildSplitPaneParser()
+{
+    _newPaneCommand = _app.add_subcommand("split-pane", "Create a new pane");
+    _addNewTerminalArgs(_newPaneCommand);
+    auto* horizontalOpt = _newPaneCommand->add_flag("-H,--horizontal",
+                                                    _splitHorizontal,
+                                                    "Create the new pane as a horizontal split (think [-])");
+    auto* verticalOpt = _newPaneCommand->add_flag("-V,--vertical",
+                                                  _splitVertical,
+                                                  "Create the new pane as a vertical split (think [|])");

```

Comment:
localization

Reply:
(I'm resolving all these loc conversations as part of the same "loc makes tests sad" convo)